### PR TITLE
Use SystemSocketFactory instead of the default factory for SSL socket factory

### PR DESCRIPTION
**Pull request by jacksierkstra**
Solves: https://github.com/GoogleContainerTools/jib/issues/2585

### CODE REVIEW

**Comment by jacksierkstra** [comment]
@googlebot I signed it!

File: jib-core/src/main/java/com/google/cloud/tools/jib/http/FailoverHttpClient.java [hunk 1]
```diff
@@ -97,7 +101,17 @@ private static HttpTransport getSecureHttpTransport() {
...
+        HttpClientBuilder.create()
+            .useSystemProperties()
+            .setSSLSocketFactory(SSLConnectionSocketFactory.getSystemSocketFactory())
+            .setMaxConnTotal(200)
+            .setMaxConnPerRoute(20)
+            .setConnectionTimeToLive(-1, TimeUnit.MILLISECONDS)
+            .setRoutePlanner(new SystemDefaultRoutePlanner(ProxySelector.getDefault()))
+            .disableRedirectHandling()
+            .disableAutomaticRetries();
+    return new ApacheHttpTransport(httpClientBuilder.build());
```
Comment (chanseokoh) [inline jib-core/src/main/java/com/google/cloud/tools/jib/http/FailoverHttpClient.java:114]: Thanks for your contribution!

As in `getInsecureHttpTransport()` below, can't this just be
```java
          ApacheHttpTransport.newDefaultHttpClientBuilder()
              .setSSLSocketFactory(SSLConnectionSocketFactory.getSystemSocketFactory());
```
? This is exactly [the reason for asking Google HTTP Client to add `newDefaultHttpClientBuilder()`](https://github.com/googleapis/google-http-java-client/issues/578) in the past.

Comment (jacksierkstra) [reply jib-core/src/main/java/com/google/cloud/tools/jib/http/FailoverHttpClient.java]: Yes, that would work and is cleaner. I applied your suggestion.

**Comment by chanseokoh** [comment]
Cool. The code is not formatted correctly. Try running `./gradlew googleJavaFormat`.

**Pull request merged**